Source/core/Sync.cpp

```diff
@@ -782,6 +784,207 @@ namespace Core {
         return (nResult);
     }
 
+    //----------------------------------------------------------------------------
+    //----------------------------------------------------------------------------
+    // SharedSemaphore class
+    //----------------------------------------------------------------------------
+    //----------------------------------------------------------------------------
+
+    //----------------------------------------------------------------------------
+    // CONSTRUCTOR & DESTRUCTOR
+    //----------------------------------------------------------------------------
+
+
+    SharedSemaphore::SharedSemaphore(const TCHAR sourceName[], const uint32_t initCount, VARIABLE_IS_NOT_USED const uint32_t maxCount)
+    {
+        ASSERT(initCount <= 1);
+        ASSERT(maxCount <= 1);
+        ASSERT(initCount <= maxCount);
+#ifdef __WINDOWS__
+        _semaphore = (::CreateSemaphore(nullptr, initCount, maxCount, sourceName));
+#else
+        _name = sourceName;
+        _semaphore = sem_open(sourceName, O_CREAT | O_RDWR, 0644,  initCount);
```

Comment:
Perhaps we should cater for creating a semaphore with the same name. I,e. differentiate between creating and opening. Now if you mistakenly ask for a semaphore with the same name you'll just open the existing one, and do not know about the fact. Same goes for Windows.